### PR TITLE
fix(ivy): ensure module imports are instantiated before the module being declared

### PR DESCRIPTION
**Pull request by atscott**

### CODE REVIEW

File: packages/core/src/di/r3_injector.ts [hunk 1]
```diff
@@ -290,7 +290,6 @@ export class R3Injector {
     }
 
     // Track the InjectorType and add a provider for it.
```
Comment (AndrewKushnir) [inline packages/core/src/di/r3_injector.ts]: nit: it looks like comment should also be updated/rephrased to reflect code change.

Comment (atscott) [reply packages/core/src/di/r3_injector.ts]: Removed.

File: packages/core/src/di/r3_injector.ts [hunk 2]
```diff
@@ -330,6 +326,9 @@ export class R3Injector {
         }
       }
     }
+    // Track the InjectorType and add a provider for it.
```
Comment (alxhub) [inline packages/core/src/di/r3_injector.ts]: Add a comment here about the ordering being important - recursing into children and _then_ adding the injectortype to the set.

File: packages/core/test/acceptance/ng_module_spec.ts [hunk 3]
```diff
@@ -77,6 +77,25 @@ describe('NgModule', () => {
     });
   });
 
+  it('initializes the module imports before the module itself', () => {
```
Comment (alxhub) [inline packages/core/test/acceptance/ng_module_spec.ts]: This test is great, but it would be nice to have a simple r3_injector_spec test as well.

**Review by alxhub — APPROVED** [review]

**Comment by atscott** [comment]
[presubmit](https://test.corp.google.com/ui#id=OCL:293702733:BASE:293703076:1581034970421:73b064f1) (failing tests passed in a rerun)

**Pull request closed without merging**